### ISSUE TITLE
[question] How to write single Files from each style within a variable font

### ISSUE THREAD

**Issue by BURROO** [opentypejs/opentype.js#483]
I am curious if it is possible to load a variable font file and generate all its associated styles as a buffer and eventually create files (otf, woff2, etc) from them. Since all the information from each of those styles is already available in the variable font this should easily be possible, right? What would be the best approach to approach this or is there a function included that helps for example with creating the dataset for each glyph of the associated style?

**Comment by yne** [comment]
Variable font are not supported by opentype.js

**Comment by BURROO** [comment]
Thank you! Seeing that you are involved in the project, do you know if there is support for variable planed eventually or could you point me to any other tools/resources that could parse/write variable fonts in a similar matter as opentype.js?

**Comment by yne** [comment]
I was personally interested in the matter, so I started reading the [OTF spec](https://docs.microsoft.com/en-us/typography/opentype/spec/otvaroverview) and tried to implement it but I got overwhelmed but the number of table needed.

Also OpentypeJS use the [CFF](https://docs.microsoft.com/en-us/typography/opentype/spec/cff) table for defining it glyph shape. But the spec explicitly state that:

> variations for outlines using the [(Compact Font Format version 1.0 ('CFF ')](https://docs.microsoft.com/en-us/typography/opentype/spec/cff) table are not supported.

So it would also require OpentypeJS to migrate to CFF2 or some less-compact table declaration to be able to use font variation feature.

But anybody could do send a PR to change that. So no ETA for now :)

No idea about other tools, but I use [ttx](https://github.com/fonttools/fonttools) to help me dump OTF to XML in order to understand them.
